cc-db-service/openapi.yaml

```diff
@@ -0,0 +1,438 @@
+              $ref: '#/components/schemas/PostUserItem'
+        required: true
+      responses:
+        '200':
+          description: 'OK'
+        '400':
+          description: Invalid request data
+        '404':
+          description: Not found
+        '500':
+          description: Service error
+  /v1/items/draft:
+    post:
+      operationId: postDraftValues
+      description: Create draft for Item and save form values
+      tags:
+        - Items
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PostDraft'
+        required: true
+      responses:
+        '200':
+          description: 'OK'
+        '400':
+          description: Invalid request data
+        '404':
+          description: Not found
+        '500':
+          description: Service error
+
+  /v1/items/draft/{draftId}:
+    put:
+      operationId: putDraftValues
+      description: Update draft for Item and save new values
+      tags:
+        - Items
+      parameters:
+        - name: draftId
+          required: true
+          in: path
+          description: Draft id
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PutDraft'
+        required: true
+      responses:
+        '200':
+          description: 'OK'
+        '400':
+          description: Invalid request data
+        '404':
+          description: Not found
+        '500':
+          description: Service error
+  /v1/configurations:
+    post:
+      operationId: postConfiguration
+      description: Create new configuration
+      tags:
+        - Configurations
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PostConfiguration'
+        required: true
+      responses:
+        '200':
+          description: 'OK'
+        '400':
+          description: Invalid request data
+        '404':
+          description: Not found
+        '500':
+          description: Service error
+  /v1/configurations/{configurationId}:
+    put:
+      operationId: putConfiguration
+      description: Update configuration
+      tags:
+        - Configurations
+      parameters:
+        - name: configurationId
+          required: true
+          in: path
+          description: Configuration id
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PutConfiguration'
+        description: payload
+        required: true
+      responses:
+        '200':
+          description: 'OK'
+        '400':
+          description: Invalid request data
+        '404':
+          description: Not found
+        '500':
+          description: Service error
+components:
```

Comment:
You could add a securityScheme for the bearerAuthentication. For example, PK service and backend are using this in the components (the dashes represent empty spaces):

securitySchemes:
--bearerAuth:
----type: http
----scheme: bearer
----bearerFormat: JWT

Reply:
True, but I just don't know how we authenticate the requests for the database queries yet so I left it out. 
The backend should check that requests that need authentication have _some_ token with them, and PK uses the token received from PK keycloak. 
And since we should only allow requests from the backend service to this database service, it might be enough that the authentication is checked in the backend service

Reply:
Got it. Just want to mention that when I checked this document on the online swagger editor tool (https://editor.swagger.io/) it's commenting about the fact that you are using a bearerAuth in the security -section, but you haven't defined a securityScheme for it. You can check it on the online editor too so you can also see the error yourself.

Swagger documentation (https://swagger.io/docs/specification/authentication/) also guides to first define a securityScheme that will then be applied in the security -section. The way I understood it was that the securityScheme is needed if you want to have something in the security -section? So the options are to either 1) define a securityScheme or 2) remove the security -section. Or maybe there is something that I missed, that's also possible :D

Reply:
removed security section